### PR TITLE
Add Email support

### PR DESCRIPTION
**Pull request by estevez-dev**

### CODE REVIEW

File: src/nestor/models/contacts_book.py [hunk 1]
```diff
@@ -34,6 +35,21 @@ def value(self, value: str):
...
+    format_regexp = r'^\S+@\S+\.\S+$'
+
+    """Class representing a email field."""
+    @property
+    def value(self):
+        return self._value 
+    
+    @value.setter
+    def value(self, value: str):
+        if re.match(Email.format_regexp, value) == None:
```
Comment (VOSolyanik) [inline src/nestor/models/contacts_book.py]: `if re.match(Email.format_regexp, value) is None:`

File: src/nestor/utils/input_error.py [hunk 2]
```diff
@@ -20,9 +20,7 @@ def inner(*args, **kwargs):
                 return func(*args, **kwargs)
             except KeyError as e:
                 return Colorizer.error(f"Contact {e} not found")
-            except PhoneValueError as e:
-                return Colorizer.error(e)
-            except BirthdayValueError as e:
+            except ContactsBookException as e:
```
Comment (VOSolyanik) [inline src/nestor/utils/input_error.py:23]: Nice

File: src/nestor/models/contacts_book.py [hunk 3]
```diff
@@ -59,9 +75,10 @@ def __init__(self, name):
         self.name = Name(name)
         self.phones = []
         self.birthday = None
+        self.email = None
 
     def __str__(self):
-        return f"Contact name: {self.name.value}, phones: {'; '.join(p.value for p in self.phones)}, birthday: {self.birthday or 'not specified'}"
+        return f"Contact name: {self.name.value}, phones: {'; '.join(p.value for p in self.phones)}, birthday: {self.birthday or 'not specified'}, email: {self.email or 'not specified'}"
```
Comment (VOSolyanik) [inline src/nestor/models/contacts_book.py]: To not duplicate 'not specified', let's move to const

File: src/nestor/models/contacts_book.py [hunk 4]
```diff
@@ -83,6 +100,24 @@ def find_phone(self, phone: str) -> Phone | None:
...
+
+        self.email = Email(new_email)
+
+    def edit_email(self, new_email: str) -> None:
+        """Edit email in record."""
+        self.email = Email(new_email)
+
+    def remove_email(self) -> None:
+        """Remove email from record."""
+        if self.email == None:
```
Comment (VOSolyanik) [inline src/nestor/models/contacts_book.py]: Let's not raise error here.
For Add contact we have the logic that we update contact if exist, and add if not.

File: src/nestor/models/contacts_book.py [hunk 5]
```diff
@@ -83,6 +100,24 @@ def find_phone(self, phone: str) -> Phone | None:
             if p.value == phone:
                 return p
         return None
+
+    def add_email(self, new_email: str) -> None:
```
Comment (VOSolyanik) [inline src/nestor/models/contacts_book.py]: Let's not raise error here.
For Add contact we have the logic that we update contact if exist, and add if not.

**Comment by estevez-dev** [comment]
Pushed the fixes

File: src/nestor/models/contacts_book.py [hunk 6]
```diff
@@ -54,14 +70,17 @@ def value(self, value: str):
             raise BirthdayValueError("Invalid date format. Use DD.MM.YYYY") from exc
     
 class Contact:
+    NOT_SPECIFIED_FIELD_VALUE = 'not specified'
```
Comment (VOSolyanik) [inline src/nestor/models/contacts_book.py]: better to create constants.py and move it there, because it could be used in other classes

**Review by VOSolyanik — APPROVED** [review]

**Pull request merged**